voidcrew/code/game/objects/items/devices/walkman/cassette.dm

```diff
@@ -0,0 +1,166 @@
+/obj/item/device/cassette_tape
+	name = "cassette Tape"
+	desc = "A cassette tape"
+	icon = 'voidcrew/icons/obj/walkman.dmi'
+	icon_state = "cassette_flip"
+	w_class = WEIGHT_CLASS_SMALL
+	//icon of the cassettes front side
+	var/side1_icon = "cassette"
+	//if the cassette is flipped, for playing second list of songs
+	var/flipped = FALSE //Tape side
+	//list of songs each side has to play
+	var/list/songs = list()
+	//list of each songs name in the order they appear
+	var/list/song_names = list()
+	//the id of the cassette
+	var/id = 1
+
+/obj/item/device/cassette_tape/attack_self(mob/user)
+	..()
+
+	if(flipped == TRUE)
+		flipped = FALSE
+		icon_state = side1_icon
+	else
+		flipped = TRUE
+		icon_state = "cassette_flip"
+	to_chat(user,"You flip [src]")
+
+/obj/item/device/cassette_tape/attackby(obj/item/item, mob/living/user)
+	if(istype(item, /obj/item/pen))
+		var/choice = input("What would you like to change?") in list("Cassette Name", "Cassette Description", "Cancel")
+		switch(choice)
+			if("Cassette Name")
+				//the name we are giving the cassette
+				var/newcassettename = reject_bad_text(stripped_input(user, "Write a new Cassette name:", name, name))
+				if(!user.canUseTopic(src, BE_CLOSE))
+					return
+				if (length(newcassettename) > 20)
+					to_chat(user, "<span class='warning'>That name is too long!</span>")
+					return
+				if(!newcassettename)
+					to_chat(user, "<span class='warning'>That name is invalid.</span>")
+					return
+				else
+					name = "[lowertext(newcassettename)]"
+			if("Cassette Description")
+				//the description we are giving the cassette
+				var/newdesc = stripped_input(user, "Write a new description:", name, desc)
+				if(!user.canUseTopic(src, BE_CLOSE))
+					return
+				if (length(newdesc) > 180)
+					to_chat(user, "<span class='warning'>That description is too long!</span>")
+					return
+				if(!newdesc)
+					to_chat(user, "<span class='warning'>That description is invalid.</span>")
+					return
+				else
+					desc = newdesc
+			else
+				return
+
+
+/obj/item/device/cassette_tape/verb/flip()
```

Comment:
We should avoid new verbs. They aren't really useful, especially not for this when attack_self already handles it